src/Task/TwigCsFixer.php

```diff
@@ -0,0 +1,118 @@
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Config\ConfigOptionsResolver;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use GrumPHP\Fixer\Provider\FixableProcessResultProvider;
+use Symfony\Component\Process\Process;
+
+/**
+ * @extends AbstractExternalTask<ProcessFormatterInterface>
+ */
+class TwigCsFixer extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): ConfigOptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'paths' => ['.'],
+            'level' => 'NOTICE',
+            'config' => null,
+            'report' => 'text',
+            'fix' => false,
+            'no-cache' => false,
+            'debug' => false,
+            'quiet' => false,
+            'version' => false,
+            'ansi' => false,
+            'no-ansi' => false,
+            'no-interaction' => false,
+            'verbose' => false,
+            'triggered_by' => ['twig'],
+        ]);
+
+        $resolver->addAllowedTypes('paths', ['array']);
+        $resolver->addAllowedTypes('level', ['string']);
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('report', ['string']);
+        $resolver->addAllowedTypes('fix', ['bool']);
+        $resolver->addAllowedTypes('no-cache', ['bool']);
+        $resolver->addAllowedTypes('debug', ['bool']);
+        $resolver->addAllowedTypes('quiet', ['bool']);
+        $resolver->addAllowedTypes('version', ['bool']);
+        $resolver->addAllowedTypes('ansi', ['bool']);
+        $resolver->addAllowedTypes('no-ansi', ['bool']);
+        $resolver->addAllowedTypes('no-interaction', ['bool']);
+        $resolver->addAllowedTypes('verbose', ['bool']);
+
+        return ConfigOptionsResolver::fromOptionsResolver($resolver);
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+        $files = $context->getFiles()
+            ->extensions($config['triggered_by'])
+            ->paths($config['paths']);
+
+        if (\count($files) === 0) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('twig-cs-fixer');
+        $arguments->add('lint');
+
+        if ($context instanceof GitPreCommitContext) {
+            $arguments->addFiles($files);
+        }
+
+        if ($context instanceof RunContext) {
+            foreach ($config['paths'] as $path) {
```

Comment:
It would be shorter to use `$arguments->addArgumentArray('%s', $config['paths'])` instead of having the foreach here.

Reply:
Way better! Done!